src/Util/Sniffs/PassedParameters.php

```diff
@@ -0,0 +1,385 @@
+     * If passed a T_STRING which is *not* a function call, the behaviour is unreliable.
+     *
+     * Will return a array with the start token pointer, end token pointer and the raw value
+     * of the parameter at a specific offset.
+     * If the specified parameter is not found, will return false.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile   The file where this token was found.
+     * @param int                         $stackPtr    The position of the T_STRING, T_VARIABLE, T_ARRAY,
+     *                                                 T_OPEN_SHORT_ARRAY, T_ISSET, T_UNSET or T_LIST token.
+     * @param int                         $paramOffset The 1-based index position of the parameter to retrieve.
+     *
+     * @return array|false
+     */
+    public static function getParameter(File $phpcsFile, $stackPtr, $paramOffset)
+    {
+        $parameters = self::getParameters($phpcsFile, $stackPtr);
+
+        if (isset($parameters[$paramOffset]) === false) {
+            return false;
+        }
+
+        return $parameters[$paramOffset];
+
+    }//end getParameter()
+
+
+    /**
+     * Count the number of parameters which have been passed.
+     *
+     * Expects to be passed the T_STRING or T_VARIABLE stack pointer for the function call.
+     * If passed a T_STRING which is *not* a function call, the behaviour is unreliable.
+     *
+     * Extra feature: If passed an T_ARRAY or T_OPEN_SHORT_ARRAY stack pointer,
+     * it will return the number of values in the array.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file where this token was found.
+     * @param int                         $stackPtr  The position of the T_STRING, T_VARIABLE, T_ARRAY,
+     *                                               T_OPEN_SHORT_ARRAY, T_ISSET, T_UNSET or T_LIST token.
+     *
+     * @return int
+     */
+    public static function getParameterCount(File $phpcsFile, $stackPtr)
+    {
+        if (self::hasParameters($phpcsFile, $stackPtr) === false) {
+            return 0;
+        }
+
+        return count(self::getParameters($phpcsFile, $stackPtr));
+
+    }//end getParameterCount()
+
+
+    /**
+     * Get the position of the double arrow within an array item.
+     *
+     * Expects to be passed the parameter start/end as retrieved via getParameters().
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being examined.
+     * @param int                         $start     Stack pointer to the start of the array item.
+     * @param int                         $end       Stack pointer to the end of the array item.
+     *
+     * @return int|false Pointer to the double arrow if this array item has an index or false otherwise.
+     * @throws \PHP_CodeSniffer\Exceptions\RuntimeException If the start or end positions are invalid.
+     */
+    public static function getDoubleArrowPosition(File $phpcsFile, $start, $end)
```

Comment:
The methods in `PassedParameters` work for both function calls as well as array declarations and the `getDoubleArrowPosition()` method expects a `$start` and `$end` token position as retrieved for an array item via the `getParameters()` method.
This makes this method very closely bound to the `getParameters()` method, which then made it a logical choice to add the method here.

Reply:
Then it's the name `PassedParameters` that is tripping me up. I wouldn't think to go looking here for util functions to work with arrays. I don't think the keys and values inside an array definition would typically be referred to as the passed parameters of the array.

I can't think of a generic term though.

What about having an Arrays util class that extends `PassedParameters`, adds the `getDoubleArrowPosition()` method, and maybe even contains wrapper methods for the other `PassedParameters` methods that have more friendly array-sounding names, like `hasValues()` instead of `hasParameters()`?

I know it sounds a bit wasteful, but I'm just thinking of a future `Arrays` util class with methods like 'getLongestKey()`. A developer needs to know to go looking in 2 places for array helper functions, but would naturally start in the `Arrays` class and not find what they are after.

Reply:
I'm also in favour of having `Array` util class, Makes it easier to find for the developer.

Reply:
PassedParameters is fine but I agree with there should be a way to get to this through an ArrayUtil class I think most libs have Array and String utility classes which wraps or extends things that are relevant.